packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFUIDelegateHostApi.m

```diff
@@ -88,6 +88,63 @@ - (void)requestMediaCapturePermissionForDelegateWithIdentifier:(FWFUIDelegate *)
                                                             decision));
                                                   }];
 }
+
+- (void)runJavaScriptAlertPanelForDelegateWithIdentifier:(FWFUIDelegate *)instance
+                                                 message:(NSString *)message
+                                                   frame:(WKFrameInfo *)frame
+                                       completionHandler:(void (^)(void))completionHandler {
+  [self runJavaScriptAlertPanelForDelegateWithIdentifier:[self identifierForDelegate:instance]
+                                                 message:message
+                                                   frame:FWFWKFrameInfoDataFromNativeWKFrameInfo(
+                                                             frame)
+                                              completion:^(FlutterError *error) {
+                                                NSAssert(!error, @"%@", error);
```

Comment:
when will error happen? should we pipe the error to dark side rather than doing assert here?

Reply:
https://github.com/flutter/packages/blob/83c2c4da97c554996c488037c7ca7c1d5a076a9b/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFUIDelegateHostApi.m#L34-L91

@hellohuanlin In this case, I just followed previous implementation. I also agree with that NSAssert can be dangerous somethime, but the error only came from pigeon generation part like follow code and I understood that it is serious situation if channel is not established.

https://github.com/flutter/packages/blob/83c2c4da97c554996c488037c7ca7c1d5a076a9b/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFGeneratedWebKitApis.m#L3094-L3098

Reply:
We are still deciding on a better alternative for using `NSAssert` in callback methods: https://github.com/flutter/flutter/issues/140410

@jsharp83 is correct that we probably won't be able to send the error back to Dart, so we need a way to handle it on the platform side.

Definitely feel free to add feedback to this issue!